### PR TITLE
Card to review column with disapproved trusted review/508

### LINKED ISSUE #508: After NYC, a previously C Card moving straight from RF to C instead of RF to R
### Description:

WHAT SHOULD HAPPEN:
If a card in the complete column (C) gets a negative review it moves back into review feedback (RF) column. After once again getting a new PR merged, the card can then be put back into the review column (R) by the learner and await reviews, including from a trusted reviewer, so that it may move once again to C when deemed competent.

WHAT IS CURRENTLY HAPPENING
It appears that if a card is in C and is moved back to RF, and then gets moved back to R again, it automatically moves to C without waiting for more reviews to happen.
This should not happen. 

### Steps to reproduce:

This was the card I found it on:
https://tilde-front-dot-umuzi-prod.nw.r.appspot.com/card/190179

### PR DESCRIPTION
**Pull request by Andy-Nkumane**
Related issues: [please specify]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have merged the develop branch into my branch and fixed any merge conflicts
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: backend/curriculum_tracking/tests/test_AgileCard.py [hunk 1]
```diff
@@ -706,6 +707,111 @@ def test_cancel_workshop_attendance(self):
...
+
+    def test_start(self):
+        card = self.card
+        card.start_project()
+        # self.assertEqual(card.recruit_project.recruit_users, card.assignees.first())
+        card_assignees = sorted([o.id for o in card.assignees.all()])
+        project_assignees = sorted([o.id for o in card.recruit_project.recruit_users.all()])
+
+        self.assertEqual(card.status, AgileCard.IN_PROGRESS)
+        self.assertEqual(card_assignees, project_assignees)
```
Comment (Samantha-Hampton) [inline backend/curriculum_tracking/tests/test_AgileCard.py]: This all mostly seems to be covered already in this other test class:

```
class start_project_Tests(TestCase):
    def setUp(self):
        bot = SocialProfileFactory(github_name=GIT_REAL_BOT_USERNAME)
        GithubOAuthTokenFactory(user=bot.user)

    def assert_users_same(self, card, project):
        card_assignees = sorted([o.id for o in card.assignees.all()])
        card_reviewers = sorted([o.id for o in card.reviewers.all()])
        project_reviewers = sorted([o.id for o in project.reviewer_users.all()])
        project_assignees = sorted([o.id for o in project.recruit_users.all()])

        self.assertEqual(card_assignees, project_assignees)
        self.assertEqual(card_reviewers, project_reviewers)

    @mock.patch("git_real.helpers.get_repo")
    @mock.patch.object(RecruitProject, "setup_repository")
    def test_start_project_creates_project_if_not_exists(self, get_repo, *_):

        card = factories.AgileCardFactory(
            status=AgileCard.READY,
            recruit_project=None,
            content_item=factories.ProjectContentItemFactory(
                project_submission_type=ContentItem.REPOSITORY
            ),
            assignees=[SocialProfileFactory().user],
        )

        self.assertIsNone(card.recruit_project)

        get_repo.return_value = {
            "full_name": "me/kiff",
            "owner": {"login": "me"},
            "ssh_url": "https://whatever.git",
            "private": True,
            "created_at": timezone.now().strftime(GITHUB_DATETIME_FORMAT),
            "archived": False,
        }

        card.start_project()

        self.assertIsNotNone(card.recruit_project)
        project = card.recruit_project

        card.refresh_from_db()
        project.refresh_from_db()

        self.assert_users_same(card=card, project=project)
        self.assertEqual(project.content_item, card.content_item)
        self.assertTrue(project.setup_repository.called)
        self.assertEqual(card.status, AgileCard.IN_PROGRESS)
        self.assertIsNotNone(project.start_time)
```

So maybe in your setUp you can already start the project and put it in review, which could make the rest of the tests a bit cleaner?

**Review by Samantha-Hampton — CHANGES_REQUESTED** [review]
Mostly looks good, see comment about removing test part already done else where

**Review by Samantha-Hampton — APPROVED** [review]
I'm having an issue with running Tilde locally on my machine, so I couldn't run the tests, but looks good to me

**Review by Rokhuda — APPROVED** [review]

**Review by sheenarbw — CHANGES_REQUESTED** [review]
There is a test script named `curriculum_tracking/tests/test_project_cards_move_when_they_should.py`

Are these tests DRY? Or do they repeat what is in that file?

If these tests do add value, please move them to the other file. 
If these tests add no value then please remove them. 

Sam reported the bug, it would be good to see if she is happy that the bug doesn't exist

**Pull request closed without merging**